### PR TITLE
Instrument search

### PR DESCRIPTION
**Pull request by AMoo-Miki**
Signed-off-by: Miki <mehranb@amazon.com>

### Description
Add capturing of `search` and `view_search_results` as metrics events
 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: assets/js/search.js [hunk 1]
```diff
@@ -83,8 +88,16 @@
...
             const controller = new AbortController();
             abortControllers.unshift(abortControllers);
+            const startTime = Date.now();
             const response = await fetch(`https://search-api.opensearch.org/search?q=${query}&v=${docsVersion}`, {signal: controller.signal});
             const data = await response.json();
+
+            recordEvent('view_search_results', {
+                search_term: query,
+                docs_version: docsVersion,
+                duration: Date.now() - startTime, results_num: data?.results?.length || 0
```
Comment (abbashus) [inline assets/js/search.js:98]: np: move `results_num` to next line.

File: assets/js/search.js [hunk 2]
```diff
@@ -73,6 +73,11 @@
         if (query.length < 3) return hideResults(true);
         if (query === lastQuery) return;
 
+        recordEvent('search', {
```
Comment (abbashus) [inline assets/js/search.js:76]: Should we make `search` and `view_search_results` as constants like `CLASSNAME_SPINNING` above?

**Review by abbashus — APPROVED** [review]
Some minor comments. Else looks good.

File: assets/js/search.js (hunk 1 above)
Comment (AMoo-Miki) [reply assets/js/search.js:98]: good eye!

File: assets/js/search.js (hunk 2 above)
Comment (AMoo-Miki) [reply assets/js/search.js:76]: Being GA's keywords that will never change, used more than once, and need not be shared with another component, I'd say it is wasteful to const them.

**Pull request merged**